pythonforandroid/bootstraps/sdl2/build/jni/src/start.c

```diff
@@ -66,6 +66,36 @@ int file_exists(const char *filename) {
   return 0;
 }
 
+/* Helper function to escape all occurances of ' as \', and all
+ * occurances of \n and \r with \\n and \\r for inline string literals.
+ * (Unix paths can contain line breaks, so to be super safe, we need this) */
+char *escape_quotes_of_inline_string(const char* inline_code_arg) {
+  char *result_buf = malloc(strlen(inline_code_arg) * 2 + 1);
+  // (buffer size is large enough for worst case of every char escaped)
+  int targetoffset = 0;
+  int sourceoffset = 0;
+  while (sourceoffset <= strlen(inline_code_arg)) {
+    if (inline_code_arg[sourceoffset] == '\'') {
+      result_buf[targetoffset] = '\\';
+      targetoffset++;
+      result_buf[targetoffset] = '\'';
+    } else if (inline_code_arg[sourceoffset] == '\n') {
+      result_buf[targetoffset] = '\\';
+      targetoffset++;
+      result_buf[targetoffset] = 'n';
+    } else if (inline_code_arg[sourceoffset] == '\n') {
```

Comment:
I didn't follow the discussion, but you probably mean `'\r'` here, right?

Reply:
Oh right. `\r` in paths are so rare I wasn't sure whether to put it in at all, but yea that's supposed to be `\r`